inference/cluster_store.py

```diff
@@ -85,10 +86,21 @@ def build_agglomerative_clusters(self, c):
         # TODO
         pass
 
-    # Function to calculate the average similarity of current clusters 
+    # Function to calculate the average similarity of current clusters
     def average_similarity(self):
-        # TODO 
-        pass
+        similarities = []
+        for i in range(self.total_cluster):
+            for j in range(i+1, self.total_cluster):
+                cluster1 = self.clusters[i]
```

Comment:
Clusters themselves are not stored as arrays of their numerical values, but entity name. See `build_kmeans_clusters`. You will have to map the entity name back to feature array with `self.entity_map` and then calculate similarity between objects.

Reply:
Thank you for clarifying that the numeric values aren't stored in the clusters! However, it seems like the implementations for build_kmeans_clusters and build_optimal_kmeans_clusters are storing the index corresponding to each entity name (the keys of the self.entity_map) rather than the entity name itself. From my understanding of what you said, the clusters should be storing the entity names (which are the values in self.entity_map). Can you please clarify this for me?

Reply:
Sorry, what I said earlier was misleading 😅 . The clusters will indeed be storing the key value in the entity map. The idea was to implement a function in the future called `display` which would somehow _render_ the current clusters either to terminal or as a visual.